static/app/views/performance/newTraceDetails/useTraceAverageTransactionDurations.tsx

```diff
@@ -0,0 +1,61 @@
+import {useMemo} from 'react';
+import type {Location} from 'history';
+
+import type {Organization} from 'sentry/types';
+import {useDiscoverQuery} from 'sentry/utils/discover/discoverQuery';
+import EventView from 'sentry/utils/discover/eventView';
+import {MutableSearch} from 'sentry/utils/tokenizeSearch';
+
+import type {TraceTree} from './traceTree';
+
+type Props = {
+  location: Location;
+  organization: Organization;
+  tree: TraceTree;
+};
+
+export const useTraceAverageTransactionDurations = ({
+  tree,
+  location,
+  organization,
+}: Props) => {
+  const transactionTitles = useMemo(
+    () => [...tree.transactionTitles],
+    [tree.transactionTitles]
+  );
+
+  const transactionProjectIDs = useMemo(
+    () => [...tree.transactionProjectIDs],
+    [tree.transactionProjectIDs]
+  );
+
+  const conditions = new MutableSearch(
+    transactionTitles.reduce<string[]>((acc, transaction, index, allTransactions) => {
+      acc.push(`transaction:"${transaction}"`);
+      if (index < allTransactions.length - 1) {
+        acc.push('OR');
```

Comment:
The IN syntax here is significantly faster and you're less likely to run into query limit issues, ie. `transaction:[1,2,3,4,5]`

Reply:
Ahh okok